zyte_common_items/pipelines.py

```diff
@@ -53,3 +54,87 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropLowProbabilityItemPipeline:
+    """This pipeline drops an item if its probability, defined in the settings,
+    is less than the specified threshold.
+
+    There are two ways that the thresholds can be defined per item type using
+    the ``ITEM_PROBABILITY_THRESHOLDS`` setting
+    - with paths to Item classes,
+    - with Item classes.
+    Additionally, for Item's classes without their own threshold, you can set a default value.
```

Comment:
```suggestion
    By default, 0.1 threshold is used, i.e. items with probabillity < 0.1 are dropped.
    Thresholds can also be defined per item type using
    the ``ITEM_PROBABILITY_THRESHOLDS`` setting
    - with paths to Item classes,
    - with Item classes.
    The default value can be changed using "default" key in 
    ``ITEM_PROBABILITY_THRESHOLDS``.
```

Reply:
@kmike I've prepared the version for description - is it ok (with some changes from yours), if your variant to use as  final version?
```
   This pipeline drops an item if its probability, defined in the settings,
    is less than the specified threshold.

    The threshold can be set using the ITEM_PROBABILITY_THRESHOLDS setting or the DEFAULT_THRESHOLD
    class constant.

    Using ITEM_PROBABILITY_THRESHOLDS offers greater flexibility, allowing you to define thresholds
    for each Item class separately or set a default threshold for all other item classes.

    Thresholds for Item classes can be defined using either the path to the Item class or
    directly using the Item classes themselves.

    If a default value (key "default") is provided in ITEM_PROBABILITY_THRESHOLDS, it will be applied
    to all Item classes without their own specified thresholds. If no default value is provided,
    the threshold for such Item classes will default to the value set in DEFAULT_THRESHOLD,
    which is 0.1 by default i.e. items with probabillity < 0.1 are dropped.

    DEFAULT_ITEM_PROBABILITY_THRESHOLD will also be used if no ITEM_PROBABILITY_THRESHOLDS setting is provided.

    The example of using ITEM_PROBABILITY_THRESHOLDS:

        from zyte_common_items import Article

        ITEM_PROBABILITY_THRESHOLDS = {
            Article: 0.2,
            "zyte_common_items.Product": 0.3,
            "default": 0.15
        }
```

Reply:
I think it's good to document the default behavior in the beginning, i.e. state very early that by default, 0.1 is used ("By default, 0.1 threshold is used, i.e. items with probabillity < 0.1 are dropped."). This is what happens if user enables the pipeline, without adding any configuration, and it's actualy a very reasonable default. 

I wouldn't mention DEFAULT_THRESHOLD class constant; this is very unlikely to be relevant for the users. 

> Thresholds for Item classes can be defined using either the path to the Item class or
    directly using the Item classes themselves.

👍 

> "default": 0.15

removing comment is fine, as it's documented early

Reply:
> If a default value (key "default") is provided in ITEM_PROBABILITY_THRESHOLDS, it will be applied
    to all Item classes without their own specified thresholds. If no default value is provided,
    the threshold for such Item classes will default to the value set in DEFAULT_THRESHOLD,
    which is 0.1 by default i.e. items with probabillity < 0.1 are dropped.

I feel it's a bit too verbose. E.g. "it will be applied to all Item classes without their own specified thresholds" doesn't seem required (users understand what default is). "If no default value is provided,
    the threshold for such Item classes will default to the value set in DEFAULT_THRESHOLD,
    which is 0.1 by default i.e. items with probabillity < 0.1 are dropped." this also doesn't seem required, if we document default threshold early, and drop DEFAULT_THRESHOLD mentions.

Reply:
Ok ,git it - this is a new variant:
```
This pipeline drops an item if its probability, defined in the settings,
    is less than the specified threshold.

    By default, 0.1 threshold is used, i.e. items with probabillity < 0.1 are dropped.

    You can customize the thresholds by using the ITEM_PROBABILITY_THRESHOLDS setting that offers
    greater flexibility, allowing you to define thresholds for each Item class separately or
    set a default threshold for all other item classes.

    Thresholds for Item classes can be defined using either the path to the Item class or
    directly using the Item classes themselves.

    The example of using ITEM_PROBABILITY_THRESHOLDS:

        from zyte_common_items import Article

        ITEM_PROBABILITY_THRESHOLDS = {
            Article: 0.2,
            "zyte_common_items.Product": 0.3,
            "default": 0.15
        }
```

Reply:
Looks good!